components/frontend/src/widgets/DivWithHTML.js

```diff
@@ -0,0 +1,32 @@
+import { createRef, useLayoutEffect, useState } from "react"
+
+import { childrenPropType } from "../sharedPropTypes"
+
+export function DivWithHTML({ children }) {
+    const minHeight = 15
+    const maxHeight = 60
+    const ref = createRef()
+    // Keep track of whether the div is overflown, so the resize handle can be hidden when not needed
+    const [overflown, setOverflown] = useState(false)
+    const [scrollHeight, setScrollHeight] = useState(minHeight)
+    useLayoutEffect(() => {
+        setOverflown(ref.current.clientHeight < ref.current.scrollHeight)
+        setScrollHeight(ref.current.scrollHeight)
```

Comment:
I don't understand how this works to adjust the `scrollHeight`, but seems like some nice js magic :)

Reply:
The `ref.current.scrollHeight` isn't available until after the size of the children prop has been calculated. So we can't pass it directly to the div style. The `useLayoutEffect` hook runs after the children prop has been rendered and the scrollHeight is known.